src/sentry/api/endpoints/organization_traces.py

```diff
@@ -495,50 +508,65 @@ def get_traces_matching_metric_conditions(
 
         return min_timestamp, max_timestamp, trace_ids
 
-    def get_traces_matching_span_conditions(
+    def get_traces_matching_span_conditions_timestamp_order(
         self,
         params: ParamsType,
         snuba_params: SnubaParams,
         trace_ids: list[str] | None = None,
     ) -> tuple[datetime, datetime, list[str]]:
-        all_queries: list[QueryBuilder] = []
-        timestamp_column: str | None = None
+        assert self.sort is not None
 
-        query, timestamp_column = self.get_traces_matching_span_conditions_query(
+        builder, timestamp_column = self.get_traces_matching_span_conditions_query(
             params,
             snuba_params,
+            orderby=[self.sort, "trace"],
         )
 
-        all_queries.append(query)
+        executor = OrderedTracesExecutor(
+            params=params,
+            snuba_params=snuba_params,
+            builder=builder,
+            limit=self.limit,
+            timestamp_column=timestamp_column,
+            max_execution_seconds=30,
+            max_parallel_queries=3,
```

Comment:
Good call on parallel to speed it up a bit, are we sure this won't spam Snuba and hit rate limits?

Reply:
3 x 8 hour queries should stay within the rate limits of snuba but I'll confirm before merging. And I explicitly moved this to a different referrer to help alleviate this concern slightly.